docs/guide/writing-tests-for-modules.md

```diff
@@ -0,0 +1,271 @@
+- and many more
+
+This documentation focuses on how to test Apostrophe modules, not Apostrophe itself using integration testing.
+
+Modules should be stand-alone npm packages. They need to be hosted somewhere that our testing can access. For this
+example, we are hosting this module in a github repo. We will use a fictitious module named `article` to illustrate how to
+write tests for modules.
+
+## Requirements
+
+- A running MongoDB server
+
+## Setup
+
+This setup assumes you will use the following packages
+
+- [apostrophe](https://www.npmjs.com/package/apostrophe): *ApostropheCMS is a full-featured, open source CMS built with Node.js that seeks to empower organizations by combining in-context editing and headless architecture in a full-stack JS environment*
+- [eslint](https://www.npmjs.com/package/eslint): *ESLint is a tool for identifying and reporting on patterns found in ECMAScript/JavaScript code*
+- [eslint-config-apostrophe](https://www.npmjs.com/package/eslint-config-apostrophe): *An ESLint configuration for ApostropheCMS core and officials modules*
+- [mocha](https://www.npmjs.com/package/mocha): *Simple, flexible, fun JavaScript test framework for Node.js & The Browser*
+```sh
+npm install --save-dev apostrophe eslint eslint-config-apostrophe mocha
+```
+
+You will need to add some new scripts to your `package.json`.
+
+<AposCodeBlock>
+  ```json
+  {
+    "name": "article",
+    "version": "1.0.0",
+    "description": "A fictitious Apostrophe module",
+    "main": "index.js",
+    "scripts": {
+      "lint": "npm run eslint",
+      "eslint": "eslint .",
+      "test": "npm run lint && mocha"
+    },
+    "keywords": [],
+    "author": "",
+    "license": "ISC"
+  }
+  ```
+  <template v-slot:caption>
+    package.json
+  </template>
+</AposCodeBlock>
+
+Add the following to `.eslintrc.json` to tell eslint to use ApostropheCMS ESLint confiuration.
+
+<AposCodeBlock>
+  ```json
+  {
+    "extends": [ "apostrophe" ]
+  }
+  ```
+  <template v-slot:caption>
+    .eslintrc.json
+  </template>
+</AposCodeBlock>
+
+When you call `npm test`, it will lint your files according to [eslint-config-apostrophe](https://www.npmjs.com/package/eslint-config-apostrophe) rules and run the tests using [mocha](https://www.npmjs.com/package/mocha).
+
+### Test folder
+
+You will need a `test/package.json` file referencing the repository URL of your module. Please replace `article` & `%article-repository-url%` with your module informations.
+
+e.g. for the module [@apostrophecms/login-hcaptcha](https://github.com/apostrophecms/login-hcaptcha) we use 
+
+> `"@apostrophecms/login-hcaptcha": "git://github.com/apostrophecms/login-hcaptcha.git"`
+> 
+
+<AposCodeBlock>
+  ```json
+  {
+    "/**": "This package.json file is not actually installed.",
+    " * ": "Apostrophe requires that all npm modules to be loaded by moog",
```

Comment:
Don't mention moog. Just say "npm modules to be loaded"

Reply:
Our v3 docs don't talk about moog, it's an internal detail for us to worry about; they talk about apostrophe modules and how they are loaded but this is no longer a "brand" we use.